### DISCUSSION TITLE
Uploading files to Azure App Service

### DISCUSSION THREAD

**Discussion by mixlplix** [oqtane/oqtane.framework#1784] [General]
Hello,

We are using the built-in Oqtane File Upload to upload Excel files to process data.  Everything works fine locally, but when we try in Azure then we get an error in the console, even though the page comes back with a success. Here's the error we get.  I figure that uploading to the Azure App Service is likely the issue because of permissions, but what are the options, can we configure Blob storage or something else to work with Oqtane?

![image](https://user-images.githubusercontent.com/1314264/141200028-c7d72bae-1cc9-4b8c-af7f-a3b0b0a588e6.png)

**Comment by sbwalker** [comment]
@mixlplix can you please check your Oqtane Event Log to determine if there is an error recorded on the server related to this upload activity? The error above is from the client-side browser tools so it does not provide any hint of what may be happening on the server.

**Reply by mixlplix** [reply]
So the log reports that it successfully uploaded.  Here are the two log events that give the relevant info.
![image](https://user-images.githubusercontent.com/1314264/141377197-f7a169df-8a34-48f3-bc2d-d2c76c9b3051.png)

![image](https://user-images.githubusercontent.com/1314264/141377264-1e69a6a5-d60a-4a46-8dad-119b3defa7a5.png)

But when I go to Azure and explore the folder where it says it copied the file, it does not appear to be there.
![image](https://user-images.githubusercontent.com/1314264/141377444-dfcaf402-b42b-4e95-a3b6-de3db19002e4.png)

**Comment by mixlplix** [comment]
Sorry, here is the other log that says where it was saved to
![image](https://user-images.githubusercontent.com/1314264/141377596-1dc31b9a-bae4-4564-b625-a71aab3fcda7.png)

**Reply by sbwalker** [reply]
@mixlplix based on the server log entry it looks like the file is being saved to C:\home\site\... does that represent a valid path in your Azure environment? 

The client log entry appears to be coming from a custom module - not the native Oqtane FileManager component?

**Comment by mixlplix** [comment]
Yes it does.  I went to the Console, and was able to navigate to the Sites/1 directory.  There was no Uploads folder, so I created one, but the file still did not save.

**Reply by sbwalker** [reply]
The missing Uploads folder is definitely the root of the problem. If you go to your File Manager in the Admin Dashboard, does the Uploads folder exist? If not it will definitely prevent uploads from succeeding as file uploading is dependent on the folder permissions defined in the File Manager... however I am not sure how it's even possible that you are attempting to upload to an Uploads folder if it does not even exist? How are you declaring the FileManager component in your client component? Are you hardcoding "Uploads" in the Folder parameter? If so, this will never work unless the Uploads folder has actually been created in Oqtane in advance with the appropriate framework permissions.

**Comment by mixlplix** [comment]
Thanks @sbwalker I figured it out.  The client-side error that is showing up is happening because the file name already exists in the Uploads folder, so we just need to add some code to handle that.  Thanks for helping me figure this out.

 

</doug>
